### PR TITLE
Add php7 support

### PR DESCRIPTION
**Pull request by mool**
I don't know if it's the best approach, but the idea is to use `phpenmod` if `php5enmod` doesn't exists. I added Ubuntu 16.04 platform to test kitchen to test the cookbook.

### CODE REVIEW

**Comment by mool** [comment]
The unit test are failing because the java agent URL has changed, but all the changes of the PHP agent are working ok.

File: providers/agent_php.rb [hunk 1]
```diff
@@ -88,14 +88,23 @@ def webserver_service
...
-  execute 'newrelic-php5enmod' do
-    command 'php5enmod newrelic'
+  cmd = Mixlib::ShellOut.new('which php5enmod')
+  cmd.run_command
+
+  enable_mod_cmd = if cmd.error?
+                     'phpenmod'
+                   else
+                     'php5enmod'
+                   end
```
Comment (Shelvak) [inline providers/agent_php.rb]: Better with :D 
`enable_mod_cmd = cmd.error?  ? 'phpenmod' : 'php5enmod'`

File: spec/unit/agent_php_spec.rb [hunk 2]
```diff
@@ -80,8 +80,8 @@
       end
     end
 
-    it 'defines newrelic-php5enmod execute block' do
-      expect(chef_run.execute('newrelic-php5enmod')).to do_nothing
+    it 'defines newrelic-phpenmod execute block' do
+      expect(chef_run.execute('newrelic-phpenmod')).to do_nothing
```
Comment (Shelvak) [inline spec/unit/agent_php_spec.rb]: Maybe a good point to test could be:
```ruby
it 'defines newrelic-phpenmod execute block with php5enmod' do
 Mixlib::ShellOut.stub(:new).with('which php5enmod').and_return(shellout)
  expect(chef_run.execute('newrelic-phpenmod')).to do_nothing # or call some action.
end

it 'defines newrelic-phpenmod execute block without php5enmod' do
  shellout.stub(:error?).and_return(true)
  Mixlib::ShellOut.stub(:new).with('which php5enmod').and_return(shellout)
  expect(chef_run.execute('newrelic-phpenmod')).to do_nothing  # or call some action.
end
```
Just to be sure that the `wich php5enmod` doesn't raise any exception.

**Comment by bassrock** [comment]
Any chance of this getting in or does it need some help?

**Comment by mool** [comment]
@djoos can you take a look to this PR? 😄

**Comment by djoos** [comment]
Hi,

thanks for the nudge!

FYI: I'll be looking at getting the java-issues PR/failing tests fixed first, so we can get some flow on the other outstanding PRs afterwards, ok?

Thanks!

**Comment by djoos** [comment]
Hi,

getting rid of php5enmod without leaving a bit of a workaround to allow the old execute_php5enmod-attribute will break backwards compatibility for those using that (to-become-deprecated) attribute either directly on the LWRP or in the recipe.

Any chance you'd like to have a look at implementing a workaround so we don't break BC?
FYI: there's a similar work-around for the deprecated api_key-attribute of the [deployment provider](https://github.com/djoos-cookbooks/newrelic/blob/master/providers/deployment.rb#L17-L21). It doesn't have a recipe, however in this case I'd say: just push in both the deprecated attribute if set and the new attribute if set in the php_agent-recipe? (which kind of makes the deprecated attribute take precedence if both set, however will also trigger the deprecation warning, so I think that is ok...)
What do you think?

Thanks in advance!
David

**Comment by mool** [comment]
Hi @djoos,

I fixed the backwards compatibility and added a deprecation message. I also modified the serverspec test to check the php module config file to support ubuntu 16.04 and php 7.0.

**Comment by djoos** [comment]
Sorry, I gave you some wrong pointers here - I've changed some things around, but PHP7 support will go out in 2.24.0...

Thanks for your contribution!

**Comment by jamescorkran** [comment]
Forgive me for the necromancy @mool but with these changes how do you now setup your roles/recipes to get new relic to use php7?  I'm a bit lost after these changes as it used to be semi set-it-forget-it.

**Pull request merged**